### PR TITLE
Change print to use __str__ and add __str__ to object

### PR DESCRIPTION
**Pull request by windelbouwman**

### CODE REVIEW

File: vm/src/objtype.rs [hunk 1]
```diff
@@ -170,6 +179,14 @@ pub fn new(typ: PyObjectRef, name: &str, bases: Vec<PyObjectRef>, dict: PyObject
     ))
 }
 
+fn type_str(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
+    // TODO: fix macro:
+    // arg_check!(vm, args, required = [(obj, Some(vm.ctx.type_type()))]);
```
Comment (OddBloke) [inline vm/src/objtype.rs]: What needs fixing here?

Comment (windelbouwman) [reply vm/src/objtype.rs]: The macro contains objtype, but we are in this place already in objtype, so it cannot find objtype::

Comment (windelbouwman) [reply vm/src/objtype.rs]: Wow, I read my own response back... does that make sense?

**Comment by OddBloke** [comment]
I think so! 👍

**Pull request merged**